### PR TITLE
Refactor CIDR logic

### PR DESCRIPTION
**Pull request by isra17**
@ark3 

I did cut out the refactor as asked in https://github.com/telepresenceio/telepresence/pull/1371#issuecomment-658364838

I also included https://github.com/telepresenceio/telepresence/pull/1373 as part of the refactor.

### CODE REVIEW

File: telepresence/outbound/cidr.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+from telepresence.runner import Runner
+from telepresence.utilities import random_name
+
+
+def covering_cidr(ips: List[str]) -> List[str]:
+    """
+    Given list of IPs, return CIDR that covers them all.
+
+    Presumes it's at least a /24.
+    """
```
Comment (ark3) [inline telepresence/outbound/cidr.py:31]: Please update the function name and doc string.

File: telepresence/outbound/cidr.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+
+    # Run script to convert --also-proxy hostnames to IPs, doing name
+    # resolution inside Kubernetes, so we get cloud-local IP addresses for
+    # cloud resources:
+    result = set(k8s_resolve(runner, remote_info, hosts_or_ips))
+    context_cache = runner.cache.child(runner.kubectl.context)
+    result.update(context_cache.lookup("podCIDRs", lambda: podCIDRs(runner)))
+    result.update(
+        context_cache.lookup("serviceCIDR", lambda: serviceCIDR(runner))
+    )
```
Comment (ark3) [inline telepresence/outbound/cidr.py:108]: Please update the lookup key to "serviceCIDRs" with an "s" at the end.

If you don't do this, then every existing user will get weird crashes from `sshuttle` until the cache entry is cleared. The old cache stored a string, the new one stores a list, and the code doesn't check that it got back what it expected.

File: telepresence/outbound/cidr.py (hunk 1 above)
Comment (isra17) [reply telepresence/outbound/cidr.py]: Done

File: telepresence/outbound/cidr.py (hunk 2 above)
Comment (isra17) [reply telepresence/outbound/cidr.py]: Right! Done

**Comment by isra17** [comment]
>Oh, could you also add newsfragments/1373.bugfix to this PR?

Done!

**Pull request merged**